pgmpy/sampling/base.py

```diff
@@ -92,7 +95,19 @@ def _reduce_marg(variable_cpd, reduce_index, sc):
             slice_[index] = sc[i]
 
         reduced_values = variable_cpd.values[tuple(slice_)]
-        marg_values = compat_fns.einsum(reduced_values, range(reduced_values.ndim), [0])
+
+        # Generate einsum string
+        ndim = reduced_values.ndim
+        input_axes = string.ascii_lowercase[:ndim]
+        output_axes = input_axes[0]
+        equation = f"{input_axes}->{output_axes}"
+
+        # Perform marginalization using einsum
+        if config.get_backend() == "torch":
+            reduced_values = torch.tensor(
+                reduced_values, dtype=config.get_dtype(), device=config.get_device()
```

Comment:
Why is `reduced_values` not already a tensor? If the backend is torch, this should already be a tensor object, if that is not the case, we will need to find the source of this issue.